src/main/java/com/appium/manager/DeviceAllocationManager.java

```
@@ -113,6 +124,43 @@ private void initializeDevices() throws Exception {
         }
     }
 
+    private void setFlagsForCapsValues() {
+        String filePath = getCapsFilePath();
+        JSONArray jsonParsedObject = new JsonParser(filePath).getJsonParsedObject();
+        Object getPlatformObject = jsonParsedObject.stream().filter(o -> ((JSONObject) o)
+                .get("iOS") != null)
+                .findFirst();
+        Object platFormCapabilities = ((JSONObject) ((Optional) getPlatformObject)
+                .get()).get("iOS");
+
+        ((JSONObject) platFormCapabilities).forEach((caps, values) -> {
+            if ("app".equals(caps)) {
+                if (values instanceof JSONObject
+                        && (((JSONObject) values).get("simulator") != null)) {
+                    simCapsPresent = true;
+                } else if (values instanceof JSONObject
+                        && (((JSONObject) values).get("device") != null)) {
+                    deviceCapsPresent = true;
+                }
+            }
+        });
+    }
+
+    private String getCapsFilePath() {
+        String filePath = appiumDriverManager.getCapsPath();
+        if (new File(filePath).exists()) {
+            Path path = FileSystems.getDefault().getPath(filePath);
+            if (!path.getParent().isAbsolute()) {
+                filePath = path.normalize()
+                        .toAbsolutePath().toString();
+            }
+            return filePath;
+        } else {
+            System.out.println("Capability file not found");
+            return null;
```

Overall looks good. Can we throw exception here instead of returning null?